lib/cli/config.js

```diff
@@ -24,6 +24,7 @@ exports.CONFIG_FILES = [
   '.mocharc.js',
   '.mocharc.yaml',
   '.mocharc.yml',
+  '.mocharc.jsonc',
   '.mocharc.json'
 ];
 
```

Comment:
Somewhat related -- asking for opinions here... not changes atm.

Think this is better?
```js
exports.loadConfig = filepath => {
  let config = {};
  const ext = path.extname(filepath);
  try {
    if (/\.ya?ml/.test(ext)) {
      config = parsers.yaml(filepath);
    } else if (ext === '.js') {
      config = parsers.js(filepath);
    } else if (/\.jsonc?/.test(ext)) {
      config = parsers.json(filepath);
    } else {
      throw new Error('no parser for this type file');
    }
  } catch (err) {
    throw new Error(`failed to parse "${filepath}": ${err}`);
  }
  return config;
};
```
Rationale: However stupid, if I used `--config` with a Word document argument, the code would throw a `SyntaxError` (as result of failing `JSON.parse`) which is utterly unhelpful and its error message doesn't
 address the actual problem.

Reply:
I agree the fallback to the JSON parser can lead to an unhelpful error message.
But in your snippet the inner throw would be hidden by the upper catch was that your intention?
Another point I noticed is that the file extension is case sensitive. Was that on purpose?

Reply:
Threw from `try` just to pass message to `catch` where it can generate higher-level `Error`.
Dislike the thought of it, but guess could throw `String` to pass along less baggage...

Reply:
> Another point I noticed is that the file extension is case sensitive. Was that on purpose?

We support lowercase file extensions.